src/features/DebugSession.ts

```diff
@@ -40,4 +40,109 @@ export class DebugSessionFeature implements IFeature {
     }
-}
\ No newline at end of file
+}
+
+interface ProcessItem extends vscode.QuickPickItem {
+	pid: string;	// payload for the QuickPick UI
+}
+
+interface PSHostProcessInfo {
+    processName: string;
+    processId: string;
+    appDomainName: string;
+    mainWindowTitle: string;
+}
+
+namespace GetPSHostProcessesRequest {
+    export const type: RequestType<any, GetPSHostProcessesResponseBody, string> =
+        { get method() { return 'powerShell/getPSHostProcesses'; } };
+}
+
+interface GetPSHostProcessesResponseBody {
+    hostProcesses: PSHostProcessInfo[];
+}
+
+export class PickPSHostProcessFeature implements IFeature {
+
+    private command: vscode.Disposable;
+    private languageClient: LanguageClient;
+    private waitingForClientToken: vscode.CancellationTokenSource;
+
+    constructor() {
+        this.command =
+            vscode.commands.registerCommand('PowerShell.PickPSHostProcess', () => {
+
+                if (!this.languageClient && !this.waitingForClientToken) {
+
+                    // If PowerShell isn't finished loading yet, show a loading message
+                    // until the LanguageClient is passed on to us
+                    this.waitingForClientToken = new vscode.CancellationTokenSource();
+                    vscode.window
+                        .showQuickPick(
+                            ["Cancel"],
+                            { placeHolder: "Select PowerShell Host Process to attach to: Please wait, starting PowerShell..." },
+                            this.waitingForClientToken.token)
+                        .then(response => { if (response === "Cancel") { this.clearWaitingToken(); } });
+
+                    // Cancel the loading prompt after 60 seconds
+                    setTimeout(() => {
+                            if (this.waitingForClientToken) {
+                                this.clearWaitingToken();
+
+                                vscode.window.showErrorMessage(
+                                    "Select PowerShell Host Process to attach to: PowerShell session took too long to start.");
+                            }
+                        }, 60000);
+                }
+                else {
+                    return this.pickPSHostProcess();
+                }
+            });
+    }
+
+    public setLanguageClient(languageClient: LanguageClient) {
+        this.languageClient = languageClient;
+
+        if (this.waitingForClientToken) {
+            this.clearWaitingToken();
+            return this.pickPSHostProcess();
+        }
+    }
+
+    public dispose() {
+        this.command.dispose();
+    }
+
+    // In node, the function returned a Promise<string> not sure about "Thenable<string>"
+	private pickPSHostProcess(): Thenable<string> {
+		return this.languageClient.sendRequest(GetPSHostProcessesRequest.type, null).then(hostProcesses => {
+			var items: ProcessItem[] = [];
+
+			for(var p in hostProcesses) {
```

Comment:
I pushed a commit to address that.  But we still have a problem.  When there are no host processes (and the command picker returns null) OR when the user presses Esc to close the quick pick UI without selecting anything, the extension continues to try to attach.  I did a quick search in the logs and apparently the JSON reader on the other side doesn't like me returning null when it expects an int:
```
    System.AggregateException: One or more errors occurred. ---> Newtonsoft.Json.JsonReaderException: Could not convert string to integer: null. Path 'processId', line 1, position 128.
       at Newtonsoft.Json.JsonReader.ReadAsInt32Internal()
       at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.ReadForType(JsonReader reader, JsonContract contract, Boolean hasConverter)
```
And when the user presses Esc instead of choosing, same thing:
```
    System.AggregateException: One or more errors occurred. ---> Newtonsoft.Json.JsonSerializationException: Error converting value {null} to type 'System.Int32'. Path 'processId', line 1, position 124. ---> System.InvalidCastException: Null object cannot be converted to a value type.
       at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
```
I'm thinking we should change the PSES Attach request args to take a string for the process id to make the JSON deserializer happy and then in the code, we can test for String.IsNullOrEmpty and abort attach in that scenario. Or does JSON handle Nullable<int>?  Is so, that would probably be simpler.  Thoughts?